### PR TITLE
fixed permission email bug

### PR DESCRIPTION
**Pull request by michaone007**

### CODE REVIEW

**Comment by michaone007** [comment]
Ready to merge

File: app/src/main/java/com/example/zakiva/santa/Models/FacebookSignUp.java [hunk 1]
```diff
@@ -64,23 +65,30 @@ public void onCompleted(JSONObject json, GraphResponse response) {
...
+                            Log.d(MainActivity.TAG, "name = " + json.getString("name"));
+                        }
+                        if(json.has("age_range")) {
+                            ageRange = json.getString("age_range");
+                        }
                         Log.d(MainActivity.TAG, "granted: " + AccessToken.getCurrentAccessToken().getPermissions());
                         Log.d(MainActivity.TAG, "granted: " + AccessToken.getCurrentAccessToken().getDeclinedPermissions());
                         Infra.updateUserAttributes("facebook",ageRange,gender,name);
-                        String formattedEmail = Infra.formatEmail(json.getString("email"));
+                        if(json.has("email")) {
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Models/FacebookSignUp.java]: we must have the email.

Comment (zakiva) [reply app/src/main/java/com/example/zakiva/santa/Models/FacebookSignUp.java]: add phone

Comment (zakiva) [reply app/src/main/java/com/example/zakiva/santa/Models/FacebookSignUp.java]: add fb id

File: app/src/main/java/com/example/zakiva/santa/Models/FacebookSignUp.java [hunk 2]
```diff
@@ -29,6 +29,7 @@
 public class FacebookSignUp extends AppCompatActivity {
 
     private CallbackManager callbackManager;
+    private String name, gender, ageRange, formattedEmail;
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Models/FacebookSignUp.java]: add = "NONE"
same in google

**Comment by michaone007** [comment]
Ready to merge

File: app/src/main/java/com/example/zakiva/santa/Helpers/Infra.java [hunk 3]
```diff
@@ -375,6 +375,9 @@ public static void updateUserAttributes(String type,String ageRange,String gende
         myDatabase.child("users").child(userEmail).child("attributes").child("gender").setValue(gender);
         myDatabase.child("users").child(userEmail).child("attributes").child("name").setValue(name);
     }
+    public static void addEmailAttributeToUser(String email){
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Helpers/Infra.java]: remove

File: app/src/main/java/com/example/zakiva/santa/Models/FacebookSignUp.java [hunk 4]
```diff
@@ -65,23 +70,32 @@ public void onCompleted(JSONObject json, GraphResponse response) {
...
                         Log.d(MainActivity.TAG, "granted: " + AccessToken.getCurrentAccessToken().getPermissions());
                         Log.d(MainActivity.TAG, "granted: " + AccessToken.getCurrentAccessToken().getDeclinedPermissions());
                         Infra.updateUserAttributes("facebook",ageRange,gender,name);
-                        String formattedEmail = Infra.formatEmail(json.getString("email"));
-                        Storage.setStringPreferences(Loader.userEmailFieldName , formattedEmail, getApplicationContext());
+                        Storage.setStringPreferences(Loader.userEmailFieldName , facebookUserId, getApplicationContext());
+                        if(json.has("email")) {
+                            email = json.getString("email");
+                            Log.d(MainActivity.TAG, "email" + json.getString("email"));
+                            Infra.addEmailAttributeToUser(email);
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Models/FacebookSignUp.java]: remove

File: app/src/main/java/com/example/zakiva/santa/Models/FacebookSignUp.java [hunk 5]
```diff
@@ -65,23 +70,32 @@ public void onCompleted(JSONObject json, GraphResponse response) {
...
+                        if(json.has("name")) {
+                            name = json.getString("name");
+                            Log.d(MainActivity.TAG, "name = " + json.getString("name"));
+                        }
+                        if(json.has("age_range")) {
+                            ageRange = json.getString("age_range");
+                        }
                         Log.d(MainActivity.TAG, "granted: " + AccessToken.getCurrentAccessToken().getPermissions());
                         Log.d(MainActivity.TAG, "granted: " + AccessToken.getCurrentAccessToken().getDeclinedPermissions());
                         Infra.updateUserAttributes("facebook",ageRange,gender,name);
```
Comment (zakiva) [inline app/src/main/java/com/example/zakiva/santa/Models/FacebookSignUp.java]: update with mail
same in google

**Comment by michaone007** [comment]
do not merge

**Pull request merged**